vault/logical_system.go

```diff
@@ -2473,6 +2478,58 @@ func (b *SystemBackend) responseWrappingUnwrap(ctx context.Context, te *logical.
 	return response, nil
 }
 
+func (b *SystemBackend) handleMetrics(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+
+	format := data.Get("format").(string)
+
+	acceptHeaders := req.Headers["Accept"]
+	if format == "prometheus" || (len(acceptHeaders) > 0 && strings.HasPrefix(acceptHeaders[0], "application/openmetrics-text")) {
+		if !b.Core.prometheusEnabled {
```

Comment:
What's the reason for having this guard? It's an ACL'd call so presumably if you're allowed to access it you should be allowed to get that data in whatever format you want.

Reply:
> I used the same logic as other telemetry backends, with it being disabled by default, unless you specify a valid retention in telemetry config

Right, but then unlike the other types you're plumbing a value all the way through core. Prometheus is just a format, I don't really see a reason to do this. If someone has access to the metrics, it seems like it ought to be able to be fetched in whatever format they want.

Reply:
Fair point, will modify to set a default value